### ISSUE TITLE
wrong window title/tab bar on macos, nightly

### ISSUE THREAD

**Issue by daviehh** [wez/wezterm#4966]
### What Operating System(s) are you seeing this problem on?

macOS

### WezTerm version

20240205-070437-39d2b6ca aarch64-apple-darwin

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Launch wezterm with `window_decorations = "INTEGRATED_BUTTONS|RESIZE"` and a custom font, the window title bar/ tab bar would be incorrectly displayed, see below:
<img width="253" alt="Screenshot 2024-02-05 at 3 32 20 PM" src="https://github.com/wez/wezterm/assets/25255906/0410640f-ed37-4404-af26-655eea7b56e2">

### To Reproduce

launch wezterm.app

### Configuration

```lua
local wezterm = require 'wezterm'

-- This table will hold the configuration.
local config = {}

if wezterm.config_builder then
	config = wezterm.config_builder()
end

config.font = wezterm.font 'MonaspiceNe Nerd Font Mono'
config.window_decorations = 'INTEGRATED_BUTTONS|RESIZE'

return config
```

### Expected Behavior

correct window title/tab bar

### Logs

Debug Overlay
wezterm version: 20240205-070437-39d2b6ca aarch64-apple-darwin
Window Environment: macOS 14.3 (23D56)
Lua Version: Lua 5.4
OpenGL: Apple M1 Pro 4.1 Metal - 88
Enter lua statements or expressions and hit Enter.
Press ESC or CTRL-D to exit
15:31:49.028 INFO wezterm_gui::termwindow > QuitApplication over here (window)
>

**Comment by wez** [comment]
I can't reproduce this. Is there more to your config file?

**Comment by daviehh** [comment]
No, that's the whole config file. It's only upon first launch, tab bar shows correctly after resizing the window or moving to external monitor.

**Comment by wez** [comment]
I didn't source and install the font you mentioned, because the `wezterm.font` line shouldn't influence the fancy tab bar.
If you comment out the `wezterm.font` line, does that change things?

**Comment by wez** [comment]
Do you have other window management software installed that might adjust the size or layout of windows?

**Comment by daviehh** [comment]
Thanks, no window manger, but the font line is necessary for the issue to appear.
With macos built-in fonts, `config.font = wezterm.font 'Monaco'` produces the issue while the font `Menlo` appears fine

**Comment by daviehh** [comment]
Thanks for your hard work! Looks fixed in `WezTerm-macos-20240206-225705-4f123a46`
